x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/action/TransportFollowStatsAction.java

```diff
@@ -85,7 +98,74 @@ protected void doExecute(
                 throw new ResourceNotFoundException("No shard follow tasks for follower indices [{}]", resources);
             }
         }
-        super.doExecute(task, request, listener);
+        super.doExecute(task, request, listener.map(this::enrichWithIndexDocumentCountLag));
+    }
+
+    private FollowStatsAction.StatsResponses enrichWithIndexDocumentCountLag(final FollowStatsAction.StatsResponses responses)
+        throws Exception {
+        // structure of mapping: follower index -> (remote cluster, leader index)
+        final Map<String, Tuple<String, String>> followerIndexToRemoteCluster = new HashMap<>();
+        responses.getStatsResponses()
+            .stream()
+            .map(FollowStatsAction.StatsResponse::status)
+            .forEach(
+                status -> followerIndexToRemoteCluster.computeIfAbsent(
+                    status.followerIndex(),
+                    ignored -> Tuple.tuple(status.getRemoteCluster(), status.leaderIndex())
+                )
+            );
+        followerIndexToRemoteCluster.forEach((followerIndex, remoteClusterAndIndex) -> {
+            final String remoteCluster = remoteClusterAndIndex.v1();
+            final String leaderIndex = remoteClusterAndIndex.v2();
+            try {
+                responses.setFollowerIndexDocumentCountLag(followerIndex, getDocumentCountLag(followerIndex, leaderIndex, remoteCluster));
+            } catch (Exception e) {
+                throw new RuntimeException(e);
+            }
+        });
+
+        return responses;
+    }
+
+    private long getDocumentCountLag(final String followerIndex, final String leaderIndex, final String remoteCluster) throws Exception {
+
+        final CompletableFuture<Long> localIndexDocCount = new CompletableFuture<>();
+        spawnDocCountRequest(transportService::getLocalNodeConnection, followerIndex, localIndexDocCount::complete);
+
+        final CompletableFuture<Long> remoteIndexDocCount = new CompletableFuture<>();
+        spawnDocCountRequest(
+            () -> transportService.getRemoteClusterService().getConnection(remoteCluster),
+            leaderIndex,
+            remoteIndexDocCount::complete
+        );
+
+        return remoteIndexDocCount.get() - localIndexDocCount.get();
+    }
+
+    private void spawnDocCountRequest(
+        final Supplier<Transport.Connection> connection,
+        final String indexName,
+        final Consumer<Long> docCount
+    ) {
+        SearchRequest countRequest = new SearchRequest(indexName);
+        countRequest.source(new SearchSourceBuilder().size(0).trackTotalHits(true));
```

Comment:
The parent issue mentions to get doc counts after refreshing the indices. I wonder if that means the action should refresh the indices before getting the doc counts? But refreshing may not always be desirable, so it could be an optional parameter to the API? Not sure.